server/src/main/java/com/google/finapp/FinAppService.java

```diff
@@ -191,10 +176,21 @@ public void getRecentTransactionsForAccount(
     ByteArray accountId = ByteArray.copyFrom(request.getAccountId().toByteArray());
     Timestamp beginTimestamp = Timestamp.fromProto(request.getBeginTimestamp());
     Timestamp endTimestamp = Timestamp.fromProto(request.getEndTimestamp());
+    if (endTimestamp.equals(Timestamp.MIN_VALUE)) {
```

Comment:
nit: Can we instead do something like:
`Timestamp endTimestamp = request.hasEndTimestamp() ? Timestamp.fromProto(request.getEndTimestamp()) : Timestamp.MAX_VALUE`

It's not immediately obvious to me that Timestamp.fromProto(request.getEndTimestamp()) would result in MIN_VALUE when the field is empty.

Reply:
This is a good point. However, I was thinking in proto3 that does not have optional fields and the hasFieldX() API. Given that we're moving to proto3 API we'd eventually need to write this in this way.